clients/python/pyproject.toml

```diff
@@ -135,3 +136,13 @@ show_column_numbers = true
 show_error_codes = true
 show_error_context = true
 ignore_missing_imports = true
+check_untyped_defs = true
+warn_unused_configs = true
+warn_redundant_casts = true
+warn_unused_ignores = true
```

Comment:
We want to clean up unused `type: ignore[*]` comments. In certain cases we have `type: ignore[something,unused-ignore]` only because certain errors are sensitive to the exact libraries installed or python version. For example, mypy may error with `import-untyped` on `import requests` if `types-requests` is not installed.

With an up-to-date environment `make lint` should generally be pretty consistent with the way CI runs mypy, but it's not running in an identical environment.